### PR TITLE
[Profile] Return subscriptions even with --allow-no-subscriptions

### LINKED ISSUE #11059: Can't sign in with the -allow-no-subscriptions parameter
**Resource Provider**
AZ CLI
az login
https://docs.microsoft.com/en-us/cli/azure/reference-index?view=azure-cli-latest#az-login

**Description of Feature or Work Requested**
Cannot sign in with the following command:
az login --service-principal -u <principal-id>-password <password>-tenant <tenant-id>--allow-no-subscriptions

If don't have the --allow-no-subscriptions parameter,  can sign in.

**Minimum API Version Required**
AZ CLI version 2.0.75

### PR DESCRIPTION
**Pull request by jiasli**
Fix #11059: `az login --allow-no-subscriptions` fails if there are subscriptions in the tenant

In `az login --allow-no-subscriptions`, only tenants with no subscriptions are listed and available subscriptions are missing from the subscription list. If the only tenant has some subscriptions, this command fails. 

This PR fixes this and honor the word "allow" so that available subscriptions are preserved and other tenants with no subscriptions are appended to the subscription list. 

To test:

```
az login --allow-no-subscriptions
```

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/_profile.py [hunk 1]
```diff
@@ -223,7 +223,8 @@ def find_subscriptions_on_login(self,
             t_list = [s.tenant_id for s in subscriptions]
```
Comment (jsntcy) [inline src/azure-cli-core/azure/cli/core/_profile.py]: Is it better to distinct t_list as there may be many dupplicated tenant_id?

Comment (jiasli) [reply src/azure-cli-core/azure/cli/core/_profile.py]: This it not necessary as `t_list` is for tenants which are not used to construct `bare_tenants`.

File: src/azure-cli-core/azure/cli/core/_profile.py [hunk 2]
```diff
@@ -342,7 +343,8 @@ def find_subscriptions_in_vm_with_msi(self, identity_id=None, allow_no_subscript
             if allow_no_subscriptions:
                 subscriptions = self._build_tenant_level_accounts([tenant])
             else:
-                raise CLIError('No access was configured for the VM, hence no subscriptions were found')
+                raise CLIError('No access was configured for the VM, hence no subscriptions were found. '
+                               "If this is expected, use '--allow-no-subscriptions' to have tenant level accesses.")
```
Comment (qwordy) [inline src/azure-cli-core/azure/cli/core/_profile.py]: access, uncountable noun

Comment (jiasli) [reply src/azure-cli-core/azure/cli/core/_profile.py]: Nice catch. Fixed.

**Pull request merged**